src/container_export.py

```diff
@@ -0,0 +1,522 @@
+        return []
+
+    container_ids = []
+    output_files = []
+
+    try:
+        container_ids = mount_all_containers(
+            path=disk_mount_dir,
+            container_mount_dir=all_containers_mount_root,
+            log_file=log_file,
+        )
+
+        if not container_ids:
+            message = "Mounting all containers did not find or mount any containers"
+            logger.warning(message)
+            log_entry(log_file, "Unable to mount all containers.")
+            return []
+
+        logger.info(
+            "Found and mounted %d containers under %s",
+            len(container_ids),
+            all_containers_mount_root,
+        )
+
+        for container_id in container_ids:
+            logger.info("Processing container ID: %s", container_id)
+            container_specific_mount_path = os.path.join(
+                all_containers_mount_root, container_id
+            )
+
+            if not os.path.isdir(container_specific_mount_path):
+                logger.warning(
+                    "Expected directory %s not found for container %s, skipping.",
+                    container_specific_mount_path,
+                    container_id,
+                )
+                continue
+
+            try:
+                output_file = create_output_file(
+                    output_path,
+                    display_name=container_id,
+                    data_type="container:export:zip",
+                    extension="zip",
+                    source_file_id=input_file.get("id"),
+                )
+                logger.info(
+                    "Created output file object for %s: %s",
+                    container_id,
+                    output_file.path,
+                )
+
+                archive_base = os.path.splitext(output_file.path)[0]
+                os.makedirs(os.path.dirname(archive_base), exist_ok=True)
+
+                shutil.make_archive(
+                    archive_base,
+                    "zip",
+                    root_dir=all_containers_mount_root,
+                    base_dir=container_id,
+                )
+                logger.info(
+                    "Successfully created archive for container %s at %s.zip",
+                    container_id,
+                    archive_base,
+                )
+                output_files.append(output_file)
+
+            except Exception as e:
+                message = f"Error during export of container {container_id}: {e}"
+                logger.error(message, exc_info=True)
+                log_entry(log_file, message)
+
+        logger.info("Finished processing all found containers")
+
+    except Exception as e:
+        message = "Error during export of all containers."
+        logger.error(message, exc_info=True)
+        log_entry(log_file, message)
+
+    finally:
+        logger.info("Starting cleanup for export_all_containers")
+        if container_ids:
+            unmount_all_command = [
```

Comment:
Seems duplicate of the _umount method your refactored at the top of file. Use that method?

Reply:
The `_unmount` function is to unmount a single mount point. Here were are trying to unmount all containers mounted within the specified mount point. Container Explorer creates subdirectories for each container and mounts them within the specified mount points i.e. `/mnt/abcdef/container_id_1`, `/mnt/abcdef/container_id_2`, etc.

This `unmount_all_command` unmounts `/mnt/abcdef/container_id_1`, `/mnt/abcdef/container_id_2` in a single operation.